File: chapitre-03/exemple-execlp.py
#! /usr/bin/python
# ------------------------------------------------------------------
# exemple-execlp.py
# Fichier d'exemple du livre "Developpement Systeme sous Linux"
# (C) 2015-2019 - Christophe BLAESS <christophe@blaess.fr>
# https://www.blaess.fr/christophe/
# ------------------------------------------------------------------

from __future__ import print_function
import os
import sys

i = 0
if len(sys.argv) == 2:
	try:
		i = int(sys.argv[1])
	except ValueError:
		print("Invalid value", sys.argv[1])
		sys.exit(1)
if i < 5:
	print("Counter =", (i+1))
	try:
		os.execlp(sys.argv[0], sys.argv[0], '{0}'.format(i+1))
	except OSError:
		print("Error, unable to run", sys.argv[0])

File: chapitre-03/exemple-execlp.c
// ------------------------------------------------------------------
// exemple-execlp.c
// Fichier d'exemple du livre "Developpement Systeme sous Linux"
// (C) 2000-2019 - Christophe BLAESS <christophe@blaess.fr>
// https://www.blaess.fr/christophe/
// ------------------------------------------------------------------

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main (int argc, char * argv [])
{
	char counter[2];
	int i;

	i = 0;
	if (argc == 2)
		sscanf(argv[1], "%d", & i);

	if (i < 5) {
		i ++;
		sprintf(counter, "%d", i);

		fprintf(stdout, "execlp(%s, %s, %s, NULL)\n",
		        argv[0], argv[0], counter);
		execlp(argv[0], argv[0], counter, (char *) NULL);
	}
	return 0;
}
